### PR TITLE
Introduce rcp_set_status_value filter in set_status(). #1328

### PR DESCRIPTION
**Pull request by pippinsplugins**
#1328

### CODE REVIEW

File: includes/class-rcp-member.php [hunk 1]
```diff
@@ -51,6 +51,7 @@ public function set_status( $new_status = '' ) {
 
 		$ret        = false;
 		$old_status = get_user_meta( $this->ID, 'rcp_status', true );
+		$new_status = apply_filters( 'rcp_set_status_value', $new_status, $this->ID, $old_status, $this );
```
Comment (DrewAPicture) [inline includes/class-rcp-member.php]: Docs? :-P

Comment (pippinsplugins) [inline includes/class-rcp-member.php]: Done!

File: includes/class-rcp-member.php [hunk 2]
```diff
@@ -51,6 +51,18 @@ public function set_status( $new_status = '' ) {
...
+		
+		/**
+		 * Filters the value of the status set on the member.
+		 *
+		 * @since 2.9.1
+		 *
+		 * @param string $new_status The new status to assign to the member.
+		 * @param int    $member_id The member ID.
+		 * @param string $old_status The previus status assigned to the member.
+		 * @param object $this Currenct instance of the RCP_Member object.
```
Comment (mindctrl) [inline includes/class-rcp-member.php]: `Previous` and `Current` are misspelled.

Comment (pippinsplugins) [inline includes/class-rcp-member.php]: Fixed.

**Comment by mindctrl** [comment]
Just noticed this is made against `master`. We should wait for #1321 to be merged before we do this one, or we can change the base branch to `release/2.9`.

**Comment by pippinsplugins** [comment]
Let's wait until `release/2.9` is merged into `master` then update it against `master`.

**Pull request merged**